spec/unit/indirector/catalog/rest_spec.rb

```diff
@@ -25,6 +25,23 @@ def catalog_response(catalog)
     expect(described_class.indirection.find(certname)).to be_a(Puppet::Resource::Catalog)
   end
 
+  it 'logs a notice when requesting a catalog' do
+    expect(Puppet).to receive(:notice).with("Requesting catalog from compiler.example.com:8140")
+
+    stub_request(:post, uri).to_return(**catalog_response(catalog))
+
+    described_class.indirection.find(certname)
+  end
+
+  it 'logs a notice when the IP address is resolvable when requesting a catalog' do
+    allow(Resolv).to receive_message_chain(:new, :getaddress).and_return('145.120.120.0')
```

Comment:
lets use RFC 5735 addresses and not surfcert ones
```suggestion
    allow(Resolv).to receive_message_chain(:new, :getaddress).and_return('192.0.2.0')
```

Reply:
Good catch, thanks!